### PR TITLE
New cloud provider CloudStack

### PR DESCRIPTION
**Pull request by 0xmilkmix**
This new role allows to provision instances on Apache CloudStack providers like Exoscale.

I initially wanted to add a role specific for Exoscale but since they are using Apache CloudStack, better make it generic :)

## Description
I added a new role `cloud-cloudstack` and modified the required files for inputs and configurations defaults.

## Motivation and Context
I got an Exoscale gift-card to try their service at the latest Lausanne DevSecOps meetup. Since they advertised the non-US feature of their business, we discussed VPN and the like. What better use than add support for their cloud for this gift card?

## How Has This Been Tested?
I only tested this CloudStack role on Exoscale. However, the only things to change would be in the user's `$HOME/.cloudstack.ini` file to specify the endpoint to use.

## Types of changes
- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by 0xmilkmix** [comment]
it looks like Docker build failed on `sudo -E apt-add-repository -y "ppa:wireguard/wireguard"" failed and exited with 1 during `. Unfortunately, I cannot restart this job for it to pass

File: roles/cloud-cloudstack/tasks/prompts.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    register: _cs_config_input
+    when:
+      - lookup('env', 'CLOUDSTACK_CONFIG') | length <= 0
+
+  - set_fact:
+      _cs_config: "{{ _cs_config_input.user_input | default(lookup('env', 'CLOUDSTACK_CONFIG'), true) | default('~/.cloudstack.ini') }}"
+
+  - pause:
+      prompt: |
+        Specify region to use in cloudstack.ini_file
```
Comment (jackivanov) [inline roles/cloud-cloudstack/tasks/prompts.yml]: `_` looks like a typo

Comment (0xmilkmix) [reply roles/cloud-cloudstack/tasks/prompts.yml]: indeed, fixed

**Comment by TC1977** [comment]
I'd suggest editing readme.MD to reflect the new cloud provider. Specifically line 17, line 32, line 207. And probably deserves a mention in the CHANGELOG.md.

**Comment by jackivanov** [comment]
I'm not sure if we will merge this soon, as it's currently restricted to Exoscale only. We need to implement a task to discover region automatically

**Comment by 0xmilkmix** [comment]
> I'm not sure if we will merge this soon, as it's currently restricted to Exoscale only. We need to implement a task to discover region automatically

I've not much time atm but I'll look into it in the following month.

**Comment by 0xmilkmix** [comment]
> I'm not sure if we will merge this soon, as it's currently restricted to Exoscale only. We need to implement a task to discover region automatically

done in latest commits :)

**Pull request merged**